src/Microsoft.TemplateEngine.Cli/PostActionProcessors/AddReferencePostActionProcessor.cs

```diff
@@ -102,7 +102,15 @@ private bool AddReference(IEngineEnvironmentSettings environment, IPostAction ac
                     Reporter.Error.WriteLine(LocalizableStrings.Generic_NoCallbackError);
                     return false;
                 }
-                referenceToAdd = Path.GetFullPath(referenceToAdd, outputBasePath);
+
+                // replace the referenced project file's name in case it has been renamed
+                Glob g = Glob.Parse(referenceToAdd);
```

Comment:
https://github.com/dotnet/templating/blob/4ba6e00c4754e0bf198be3aa460e34253e11d933/src/Microsoft.TemplateEngine.Cli/PostActionProcessors/PostActionProcessorBase.cs#L36

I would suggest using `GetTargetForSource` method for consistency. It does almost the same as in change

Reply:
Also `FirstOrDefault` imo is not optimal here. I would process all the files in case when several files match (but it is not really related to the PR).

Reply:
I changed the implementation to use `GetTargetForSource`, thank you for the suggestion. I used `SingleOrDefault` instead of `FirstOrDefault` as I was unsure what would be the best way to handle multiple matches.